yoti-sdk-impl/src/main/java/com/yoti/api/client/docs/DocScanService.java

```diff
@@ -0,0 +1,260 @@
+        } catch (Exception ex) {
+            throw new DocScanException("Error retrieving the session: " + ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * Deletes a session and all of it's associated content
+     *
+     * @param appId     the app ID
+     * @param keyPair   the {@code KeyPair}
+     * @param sessionId the session ID
+     * @throws DocScanException if there was an error
+     */
+    public void deleteSession(String appId, KeyPair keyPair, String sessionId) throws DocScanException {
+        notNullOrEmpty(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNullOrEmpty(sessionId, "sessionId");
+
+        String path = unsignedPathFactory.createYotiDocsSessionPath(appId, sessionId);
+        LOG.info("Deleting session from '{}'", path);
+
+        try {
+            SignedRequest signedRequest = getSignedRequestBuilder()
+                    .withKeyPair(keyPair)
+                    .withBaseUrl(apiUrl)
+                    .withEndpoint(path)
+                    .withHttpMethod(HTTP_DELETE)
+                    .build();
+
+            signedRequest.execute();
+        } catch (GeneralSecurityException ex) {
+            throw new DocScanException("Error signing the request: " + ex.getMessage(), ex);
+        } catch (ResourceException ex) {
+            throw new DocScanException("Error executing the DELETE: " + ex.getMessage(), ex);
+        } catch (IOException ex) {
+            throw new DocScanException("Error building the request: " + ex.getMessage(), ex);
+        } catch (Exception ex) {
+            throw new DocScanException("Error deleting the session: " + ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * Retrieves {@link Media} content for a given session and media ID
+     *
+     * @param appId     the app ID
+     * @param keyPair   the {@code KeyPair}
+     * @param sessionId the session ID
+     * @param mediaId   the media ID
+     * @return the {@code Media} content
+     * @throws DocScanException if there was an error
+     */
+    public Media getMediaContent(String appId, KeyPair keyPair, String sessionId, String mediaId) throws DocScanException {
+        notNullOrEmpty(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNullOrEmpty(sessionId, "sessionId");
+        notNullOrEmpty(mediaId, "mediaId");
+
+        String path = unsignedPathFactory.createMediaContentPath(appId, sessionId, mediaId);
+        LOG.info("Fetching media from '{}'", path);
+
+        try {
+            SignedRequest signedRequest = getSignedRequestBuilder()
+                    .withKeyPair(keyPair)
+                    .withBaseUrl(apiUrl)
+                    .withEndpoint(path)
+                    .withHttpMethod(HTTP_GET)
+                    .build();
+            SignedRequestResponse response = signedRequest.execute();
+            return new MediaValue(findContentType(response), response.getResponseBody());
```

Comment:
Could this return specific media types based on mime type in other SDKs, e.g. `Yoti\Media\Image\Jpeg` in PHP, or should we standardise and always return a generic _"Media"_ object in all SDKs?

Reply:
So we could do this, but I feel like we might be backing ourselves into a corner if we do this for Doc Scan where we could have many different types of Media returned.  I think for Doc Scan, returning a generic MediaValue object is the best approach as we can let them use the mimeType and do what they want with it.  Thoughts @davidgrayston ?

Reply:
Happy to always return _Media_ object, just wanted to check and follow suit in other SDKs 👍